branchLayoutApi/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -51,4 +54,49 @@ public IBranchLayout slideOut(String branchName) throws BranchLayoutException {
       return List.of(entry.withChildren(children.flatMap(child -> slideOut(child, entryToSlideOut))));
     }
   }
+
+  @Override
+  public IBranchLayout slideIn(String aboveBranchName, String branchName) throws BranchLayoutException {
+    var entry = findEntryByName(branchName).getOrNull();
+    if (entry != null) {
+      throw new BranchLayoutException("Child branch entry '${branchName}' already exists");
+    }
+    var parentEntry = findEntryByName(aboveBranchName).getOrNull();
+    if (parentEntry == null) {
+      throw new BranchLayoutException("Parent branch entry '${aboveBranchName}' does not exists");
+    }
+    var entryToSlideIn = new SlidInEntry(branchName, null, List.empty());
+    return new BranchLayout(rootEntries.flatMap(rootEntry -> slideIn(rootEntry, entryToSlideIn, parentEntry)));
+  }
+
+  @SuppressWarnings("interning:not.interned") // to allow for `entry == entryToSlideOut`
+  private List<IBranchLayoutEntry> slideIn(
+      IBranchLayoutEntry entry,
+      IBranchLayoutEntry entryToSlideIn,
+      IBranchLayoutEntry parent) {
+    var children = entry.getChildren();
+    if (entry == parent) {
+      return List.of(entry.withChildren(entry.getChildren().append(entryToSlideIn)));
+    } else {
+      return List.of(entry.withChildren(children.flatMap(child -> slideIn(child, entryToSlideIn, parent))));
+    }
+  }
+
+  @AllArgsConstructor
+  @SuppressWarnings("interning:not.interned") // to allow for `==` comparison in Lombok-generated `withChildren` method
+  private static final class SlidInEntry implements IBranchLayoutEntry {
```

Comment:
Hmm how about `@ToString` for this one? :thinking:

Reply:
Since after adding `@ToString` will be basically identical to `com.virtuslab.branchlayout.impl.BranchLayoutEntry`... possibly `BranchLayoutEntry` should go into API, just as `BranchLayout` did recently